staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -440,22 +433,44 @@ func isDeletionDup(a, b *Delta) *Delta {
 // queueActionLocked appends to the delta list for the object.
 // Caller must lock first.
 func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) error {
+	return f.queueReplaceActionLocked(actionType, false, obj)
+}
+
+// queueReplaceActionLocked appends to the delta list for the object.
+// Caller must lock first.
+func (f *DeltaFIFO) queueReplaceActionLocked(actionType DeltaType, isReplace bool, obj interface{}) error {
```

Comment:
it's confusing to have queueReplaceAction take an `isReplace bool` param... if I understand correctly, we need this so Replace can indicate the internal action type even when emitDeltaTypeReplaced is false?

if so, would something like this be clearer?

```diff
diff --git a/staging/src/k8s.io/client-go/tools/cache/delta_fifo.go b/staging/src/k8s.io/client-go/tools/cache/delta_fifo.go
index 7160bb1ee72..5ab888d7c4d 100644
--- a/staging/src/k8s.io/client-go/tools/cache/delta_fifo.go
+++ b/staging/src/k8s.io/client-go/tools/cache/delta_fifo.go
@@ -440,22 +440,37 @@ func isDeletionDup(a, b *Delta) *Delta {
 // queueActionLocked appends to the delta list for the object.
 // Caller must lock first.
 func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) error {
+	return f.queueInternalActionLocked(actionType, actionType, obj)
+}
+
+// queueActionLocked appends to the delta list for the object.
+// The actionType is emitted, and must honor emitDeltaTypeReplaced.
+// The internalActionType is only used within this function, and must ignore emitDeltaTypeReplaced.
+// Caller must lock first.
+func (f *DeltaFIFO) queueInternalActionLocked(actionType, internalActionType DeltaType, obj interface{}) error {
 	id, err := f.KeyOf(obj)
 	if err != nil {
 		return KeyError{obj, err}
 	}
 
	...

 	if f.transformer != nil {
-		var err error
-		obj, err = f.transformer(obj)
-		if err != nil {
-			return err
+		_, isTombstone := obj.(DeletedFinalStateUnknown)
+		isSyncAction := internalActionType == Sync
 		 	...
 		}
 	}
 
@@ -638,7 +653,7 @@ func (f *DeltaFIFO) Replace(list []interface{}, _ string) error {
 			return KeyError{item, err}
 		}
 		keys.Insert(key)
-		if err := f.queueActionLocked(action, item); err != nil {
+		if err := f.queueInternalActionLocked(action, Replaced, item); err != nil {
 			return fmt.Errorf("couldn't enqueue object: %v", err)
 		}
 	}
```